engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/sink/BaseKafkaAvroSinkFactory.scala

```diff
@@ -2,45 +2,46 @@ package pl.touk.nussknacker.engine.avro.sink
 
 import cats.data.Validated
 import cats.data.Validated.{Invalid, Valid}
-import org.apache.avro.Schema
-import pl.touk.nussknacker.engine.api.context.ProcessCompilationError
 import pl.touk.nussknacker.engine.api.context.ProcessCompilationError.{CustomNodeError, NodeId}
-import pl.touk.nussknacker.engine.api.process.{ProcessObjectDependencies, SinkFactory}
+import pl.touk.nussknacker.engine.api.process.SinkFactory
 import pl.touk.nussknacker.engine.api.typed.CustomNodeValidationException
 import pl.touk.nussknacker.engine.api.typed.typing.TypingResult
 import pl.touk.nussknacker.engine.api.{LazyParameter, MetaData}
+import pl.touk.nussknacker.engine.avro.AvroSchemaDeterminer
 import pl.touk.nussknacker.engine.avro.KafkaAvroFactory.{SchemaVersionParamName, SinkOutputParamName}
+import pl.touk.nussknacker.engine.avro.schemaregistry.SchemaRegistryProvider
 import pl.touk.nussknacker.engine.avro.typed.AvroSchemaTypeDefinitionExtractor
-import pl.touk.nussknacker.engine.avro.{KafkaAvroFactory, KafkaAvroSchemaProvider}
 import pl.touk.nussknacker.engine.flink.api.process.FlinkSink
-import pl.touk.nussknacker.engine.kafka.{KafkaConfig, KafkaUtils, PreparedKafkaTopic}
+import pl.touk.nussknacker.engine.kafka.{KafkaConfig, PreparedKafkaTopic}
 
 abstract class BaseKafkaAvroSinkFactory extends SinkFactory {
 
   override def requiresOutput: Boolean = false
 
   // We currently not using nodeId but it is here in case if someone want to use in their own concrete implementation
   protected def createSink(preparedTopic: PreparedKafkaTopic,
+                           version: Option[Int],
                            output: LazyParameter[AnyRef],
                            kafkaConfig: KafkaConfig,
-                           kafkaAvroSchemaProvider: KafkaAvroSchemaProvider[_],
+                           schemaRegistryProvider: SchemaRegistryProvider[_],
```

Comment:
We should pass there only serializationFactory. SchemaRegistryProvider is specific implementation. In some case for examples FixedSchema, there will be no SchemaRegistryProvider.. The same case is in Source.